### PR TITLE
INS-2867: added request reference to contract response

### PR DESCRIPTION
**Pull request by pol9kov**
**- What I did**
Added field request reference
**- How I did it**
Added field request reference to contract response
**- How to verify it**
Use simplerequester and observer to check request
**- Description for the changelog**

requestReference

### CODE REVIEW

File: insolar/messagebus.go [hunk 1]
```diff
@@ -124,6 +124,12 @@ type Reply interface {
 	Type() ReplyType
 }
 
+// ReplyWithReference include reply and request reference
+type ReplyWithReference struct {
```
Comment (LinnikD) [inline insolar/messagebus.go]: delete it

Comment (pol9kov) [reply insolar/messagebus.go]: Thank you

File: api/call.go [hunk 2]
```diff
@@ -107,20 +107,20 @@ func (ar *Runner) makeCall(ctx context.Context, request requester.Request, rawBo
...
+	result, contractErr, err := extractor.CallResponse(res.Reply.(*reply.CallMethod).Result)
 
 	if err != nil {
-		return nil, errors.Wrap(err, "[ makeCall ] Can't extract response")
+		return nil, nil, errors.Wrap(err, "[ makeCall ] Can't extract response")
 	}
 
 	if contractErr != nil {
-		return nil, errors.Wrap(errors.New(contractErr.S), "[ makeCall ] Error in called method")
+		return nil, nil, errors.Wrap(errors.New(contractErr.S), "[ makeCall ] Error in called method")
```
[the remark this answers is not in the document]
Comment (pol9kov) [reply api/call.go]: Thank you

File: api/call.go [hunk 3]
```diff
@@ -108,20 +108,20 @@ func (ar *Runner) makeCall(ctx context.Context, request requester.Request, rawBo
...
 	if err != nil {
-		return nil, errors.Wrap(err, "[ makeCall ] Can't send request")
+		return nil, nil, errors.Wrap(err, "[ makeCall ] Can't send request")
 	}
 
 	result, contractErr, err := extractor.CallResponse(res.(*reply.CallMethod).Result)
 
 	if err != nil {
-		return nil, errors.Wrap(err, "[ makeCall ] Can't extract response")
+		return nil, nil, errors.Wrap(err, "[ makeCall ] Can't extract response")
```
Comment (LinnikD) [inline api/call.go]: why requestRef nil?

File: api/call.go [hunk 4]
```diff
@@ -108,20 +108,20 @@ func (ar *Runner) makeCall(ctx context.Context, request requester.Request, rawBo
 	)
 
 	if err != nil {
-		return nil, errors.Wrap(err, "[ makeCall ] Can't send request")
+		return nil, nil, errors.Wrap(err, "[ makeCall ] Can't send request")
```
Comment (LinnikD) [inline api/call.go]: Is here always requestId nil?

Comment (pol9kov) [reply api/call.go]: Done

File: api/call.go (hunk 3 above)
Comment (pol9kov) [reply api/call.go]: Done

**Pull request merged**